lumen/sources/snowflake.py

```diff
@@ -0,0 +1,169 @@
+from __future__ import annotations
+
+from typing import Any
+
+import pandas as pd
+import param
+import snowflake.connector
+
+from ..transforms.sql import (
+    SQLDistinct, SQLFilter, SQLLimit, SQLMinMax,
+)
+from ..util import get_dataframe_schema
+from .base import BaseSQLSource, cached, cached_schema
+
+
+class SnowflakeSource(BaseSQLSource):
+    """
+    SnowflakeSource uses the snowflake-python-connector library to load data
+    from Snowflake.
+    """
+
+    account = param.String(default=None, doc="""
+        The account identifier to connect to.""")
+
+    authenticator = param.Selector(default=None, objects=['externalbrowser', 'oauth'], doc="""
+        The authentication approach to use.""")
+
+    database = param.String(default=None, doc="""
+        The database to connect to.""")
+
+    host = param.String(default=None, doc="""
+        The host to authenticate with.""")
+
+    token = param.String(default=None, doc="""
+        The OAuth token if authenticator is set to "oauth".""")
+
+    user = param.String(default=None, doc="""
+        The user to authenticate as.""")
+
+    schema = param.String(default=None, doc="""
+        The database schema to load data from.""")
+
+    warehouse = param.String(default=None, doc="""
+        The warehouse to connect to.""")
+
+    filter_in_sql = param.Boolean(default=True, doc="""
+        Whether to apply filters in SQL or in-memory.""")
+
+    sql_expr = param.String(default='SELECT * FROM {table}', doc="""
+        The SQL expression to execute.""")
+
+    tables = param.ClassSelector(class_=(list, dict), doc="""
+        List or dictionary of tables.""")
+
```

Comment:
Few if any of these parameters seem specific to Snowflake compared to any other authenticated SQL database. I guess it's fine to define them here for now, and move the shared parameters to a base class once support for the next type authenticated SQL is added? I actually can't detect any code in this class that is specific to Snowflake, so maybe this class is actually just a prototype for AuthenticatedSQLSource, tested on Snowflake so far but potentially usable for other systems as well?

Reply:
We might want to consider using SQLAlchemy as the interface to specific SQL engines.

Reply:
I'm not convinced an `AuthenticatedSQLSource` will ever really make sense. Authentication mechanisms will differ between different providers, so maybe `user` and `password` could be universal but things like, authenticator, account, token, warehouse etc. are probably not universal.

Reply:
Sounds like it's fine for now and a shared superclass can be created once we observe what's universal in practice.